### PR TITLE
Added InitialPressMouseButton to PointerReleasedEventArgs

### PR DESCRIPTION
**Pull request by kekekeks**
Added InitialPressMouseButton to PointerReleasedEventArgs that returns the button that caused the PointerPressed event.

MouseButton is now deprecated with error, since people need to decide which behavior they want

This allows people to write WPF-style code with a check for "released" mouse button. E. g. with event sequence like
- left pressed
- right pressed
- left released 
- right released

`PointerUpdateKind` will be `RightButtonReleased` while `InitialPressMouseButton` will be `MouseButton.Left`.

Also see https://github.com/AvaloniaUI/Avalonia/issues/2857#issuecomment-541293233

### CODE REVIEW

File: src/Avalonia.Input/PointerEventArgs.cs [hunk 1]
```diff
@@ -124,23 +135,29 @@ public class PointerPressedEventArgs : PointerEventArgs
...
         }
 
-        [Obsolete("Use PointerUpdateKind")] 
-        public MouseButton MouseButton => Properties.GetObsoleteMouseButton();
+        /// <summary>
+        /// Gets the mouse button that triggered the corresponding PointerPressed event
+        /// </summary>
+        public MouseButton InitialPressMouseButton { get; }
+
+        [Obsolete("Either use GetCurrentPoint(this).Properties.PointerUpdateKind or InitialPressMouseButton, see ", true)]
```
Comment (grokys) [inline src/Avalonia.Input/PointerEventArgs.cs]: `, see` at the end should be removed I think?

**Comment by grokys** [comment]
Just to check I'm understanding correctly: #2857 says that we're going to change this to work like UWP.

However  `InitialPressMouseButton`  isn't from UWP, it's instead intended as a way for people who are wanting WPF-style click events? Are we giving up on following UWP here?

**Comment by kekekeks** [comment]
>#2857 says that we're going to change this to work like UWP.

It turned out that we were already matching UWP behavior *which is different from their documentation*. See recent comments to that issue.

On the other hand there were lots of complaints about that API. So I've introduced  
`InitialPressMouseButton` as an API extension that somewhat mimics the WPF behavior by providing the information about the mouse button that caused the PointerPressed event which is useful for controls that don't need to overcomplicate things by storing pointer states in a field.

Our `Button` control, for example, doesn't really care if user was messing with the buttons, it's only interested in the left mouse button triggering the initial pointer press and the pointer still being in the bounds.

The original `MouseButton` property is deprecated with error because people need to decide what they actually need: the actual button that was released last (`e.Properties.PointerUpdateKind`), or the button that has initially triggered the event (`InitialPressMouseButton`

**Review by danwalmsley — APPROVED** [review]
LGTM

**Pull request merged**